askama_shared/src/lib.rs

```diff
@@ -327,12 +327,18 @@ struct RawEscaper<'a> {
     extensions: Vec<&'a str>,
 }
 
-fn read_config_file() -> std::result::Result<String, CompileError> {
+fn read_config_file(config_path: &Option<String>) -> std::result::Result<String, CompileError> {
     let root = PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap());
-    let filename = root.join(CONFIG_FILE_NAME);
+    let filename = if let Some(config_path) = config_path {
```

Comment:
I feel like a `match` expression would make this easier to read. (And please add a newline after it.)